ocaml/lambda/lambda.ml

```diff
@@ -1675,6 +1709,14 @@ let primitive_result_layout (p : primitive) =
       layout_boxedint Pint64
   | Pstring_load_128 _ | Pbytes_load_128 _ | Pbigstring_load_128 _ ->
       layout_boxed_vector (Pvec128 Int8x16)
+  | Pfloatarray_load_128 _ | Pfloat_array_load_128 _
+  | Punboxed_float_array_load_128 _ ->
+    layout_boxed_vector (Pvec128 Float64x2)
+  | Pint_array_load_128 _ | Punboxed_int64_array_load_128 _
+  | Punboxed_nativeint_array_load_128 _ ->
+    layout_boxed_vector (Pvec128 Int64x2)
```

Comment:
This line probably deserves a comment/CR/assertion that a `nativeint` has size `64`

Reply:
Is it possible to check `Arch.size_int` in lambda? I didn't see an obvious way. 
In any case, 128-bit operations are only supported in the x86_64 backend, so this should always be true.

Reply:
Probably not, but you could check this in `lambda_to_flambda_primitives.ml`, with the `Arch.size_int` value having been passed through from `flambda2.ml`.  (The latter file, together with the `To_cmm` files, have access to `ocamloptcomp` and hence `Arch`, whereas the remainder of Flambda 2 does not.)

Reply:
Added